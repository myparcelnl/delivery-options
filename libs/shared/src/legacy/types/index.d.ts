import {type Vue} from 'vue/types/vue';

/**
 * @deprecated
 * @deprecated
 */
declare namespace MyParcel {
  /** @deprecated */
  type Environment = 'dev' | 'staging' | 'acceptance' | 'prod';

  /** @deprecated */
  type CarrierName = 'postnl' | 'bpost' | 'dpd' | 'dhl' | 'ups' | 'dhlforyou' | 'dhlparcelconnect' | 'dhleuroplus';

  /** @deprecated */
  type CarrierID = 1 | 2 | 4 | 6 | 8 | 9 | 10 | 11;

  /** @deprecated */
  type CarrierNameOrId = CarrierName | CarrierID;

  /** @deprecated */
  type Platform = 'myparcel' | 'belgie' | 'flespakket';

  /** @deprecated */
  type CarrierIdentifier = CarrierName | `${CarrierName}:${number}`;

  /**
   * @deprecated
   * @see https://myparcelnl.github.io/api/#6_A_1
   */
  type PackageType = 'package' | 'mailbox' | 'digital_stamp' | string;

  /**
   * @deprecated
   * @see https://myparcelnl.github.io/api/#6_A_2
   */
  type DeliveryType = 'morning' | 'standard' | 'evening' | 'pickup';

  /**
   * @deprecated
   * @see https://myparcelnl.github.io/api/#6_A_3
   */
  type ShipmentOptionName =
    | 'cooled_delivery'
    | 'large_format'
    | 'only_recipient'
    | 'signature'
    | 'return'
    | 'same_day_delivery';
}

/** @deprecated */

declare namespace MyParcelDeliveryOptions {
  /** @deprecated */
  type DeliveryType = MyParcel.DeliveryType | 'sameDay';

  /**
   * @deprecated
   * Configuration object supplied by the platform.
   */
  /** @deprecated */
  interface Configuration {
    address: Address;
    config: Config;
    strings: Strings;
  }

  /**
   * @deprecated
   * Address object from the external platform.
   */
  /** @deprecated */
  interface Address {
    cc: string;
    city?: string;
    number: string | number;
    postalCode: string;
  }

  /**
   * @deprecated
   * Strings object from the external platform.
   */
  /** @deprecated */
  interface Strings {
    addressNotFound?: string;
    // BE only
    beDeliveryStandardTitle?: string;
    beDeliveryTitle?: string;
    cc?: string;
    city?: string;
    closed?: string;
    deliveryEveningTitle?: string;
    deliveryMorningTitle?: string;
    deliveryStandardTitle?: string;
    deliveryTitle?: string;
    /**
     * @deprecated
     * @see src/config/errorConfig.js
     */
    error3212?: string;
    error3224?: string;
    error3505?: string;
    error3506?: string;
    error3728?: string;
    free?: string;
    from?: string;
    headerDeliveryOptions?: string;
    loadMore?: string;
    // NL only
    mondayDeliveryTitle?: string;
    number?: string;
    onlyRecipientTitle?: string;
    openingHours?: string;
    options?: string;
    packageTypeDigitalStamp?: string;
    packageTypeMailbox?: string;
    pickUp?: string;
    pickUpFrom?: string;
    pickupLocationsListButton?: string;
    pickupLocationsMapButton?: string;
    pickupTitle?: string;
    postalCode?: string;
    retry?: string;
    saturdayDeliveryTitle?: string;
    signatureTitle?: string;
    street?: string;
    wrongPostalCodeCity?: string;
    wrongnumberPostalCode?: string;
  }

  /**
   * @deprecated
   * Response from /delivery_options
   */
  /** @deprecated */
  interface DeliveryOption {
    date: Timestamp;
    possibilities: DeliveryPossibility[];
  }

  /**
   * @deprecated
   * Response from /pickup_locations
   */
  /** @deprecated */
  interface PickupLocation {
    address: {
      cc: string;
      city: string;
      number: string;
      postal_code: string;
      street: string;
    };
    location: {
      distance: string;
      latitude: string;
      location_code: string;
      location_name: string;
      longitude: string;
      phone_number: string;
      retail_network_id: string;
      opening_hours: {
        monday: StartEndDate[];
        tuesday: StartEndDate[];
        wednesday: StartEndDate[];
        thursday: StartEndDate[];
        friday: StartEndDate[];
        saturday: StartEndDate[];
        sunday: StartEndDate[];
      };
    };
    possibilities: PickupPossibility[];
  }

  /** @deprecated */
  interface FilterableOption {
    allow: boolean;
    items: string[];
  }

  /**
   * @deprecated
   * Configuration object from the external platform.
   */
  /** @deprecated */
  interface Config {
    allowDeliveryOptions?: boolean | FilterableOption;
    allowPickupLocations?: boolean | FilterableOption;
    // Feature toggles
    allowRetry?: boolean;
    allowShowDeliveryDate?: boolean;
    apiBaseUrl?: string;
    carrierSettings?: CarrierSettings;
    currency?: string;
    cutoffTime?: string;
    deliveryDaysWindow?: string | number;
    dropOffDays?: string;
    dropOffDelay?: string | number;
    locale?: string;
    // NL only
    mondayCutoffTime?: string;
    packageType?: MyParcel.PackageType;
    pickupLocationsDefaultView?: 'map' | 'list';
    // Can be JSON string or object.
    pickupLocationsMapTileLayerData?: string | MapTileLayerData;
    pickupShowDistance?: boolean;
    platform?: MyParcel.Platform;
    // BE only
    saturdayCutoffTime?: string;
  }

  /** @deprecated */
  type CarrierSettings = {
    [key in MyParcel.CarrierIdentifier]?: {
      allowEveningDelivery?: boolean;
      allowMorningDelivery?: boolean;
      allowOnlyRecipient?: boolean;
      allowSignature?: boolean;
      allowShowDeliveryDate?: boolean;

      // NL only
      allowSaturdayDelivery?: boolean;

      // BE only
      allowMondayDelivery?: boolean;

      priceEveningDelivery?: number;
      priceMorningDelivery?: number;
      priceOnlyRecipient?: number;
      pricePickup?: number;
      priceSameDayDelivery?: number;
      priceSignature?: number;
      priceStandardDelivery?: number;
    };
  };

  /** @deprecated */
  interface CarrierData {
    deliverCountries?: string[];
    deliveryEnabled?: boolean;
    id: MyParcel.CarrierID;
    identifier: MyParcel.CarrierIdentifier;
    image: string;
    label: string;
    name: MyParcel.CarrierName;
    pickupCountries?: string[];
    pickupEnabled?: boolean;
  }

  /** @deprecated */
  interface Timestamp {
    date: string;
    timezone: string;
    timezone_type: number;
  }

  /**
   * @deprecated
   * A start and end date object.
   */
  /** @deprecated */
  interface StartEndDate {
    end: Timestamp;
    start: Timestamp;
  }

  /** @deprecated */
  interface EnumSchema<Type> {
    enum: Type[];
    type: Type extends boolean ? 'boolean' : string;
  }

  /** @deprecated */
  interface ShipmentOption {
    name: MyParcel.ShipmentOptionName;
    schema: EnumSchema<boolean>;
  }

  /** @deprecated */
  interface DeliveryTimeFrame<Type = 'start' | 'end'> {
    date_time: Timestamp;
    type: Type;
  }

  /** @deprecated */
  interface DeliveryPossibility {
    collect_date?: any;
    delivery_time_frames: [DeliveryTimeFrame<'start'>, DeliveryTimeFrame<'end'>];
    package_type: string;
    shipment_options: ShipmentOption[];
    type: MyParcel.DeliveryType;
  }

  /** @deprecated */
  interface PickupPossibility {
    delivery_type_id: number;
    delivery_type_name: MyParcel.DeliveryType;
    moment: {
      start: Timestamp;
    };
  }

  /** @deprecated */
  interface FormConfig {
    enabled?: string;
    label?: string;
    name: string;
    options?: FormConfig[];
    price?: string;
    selected?: boolean;
  }

  /** @deprecated */
  interface FormEntry {
    choices?: FormEntryChoice[];
    component?: Vue;
    dependency?: FormEntryDependency;
    hidden?: boolean;
    loop?: boolean;
    name: string;
    pagination?: number;
    type?: 'radio' | 'select' | 'checkbox' | 'text' | string;
  }

  /** @deprecated */
  interface FormEntryChoice {
    disabled?: boolean;
    label?: string;
    name: string;
    plainLabel?: string;
    price?: number;
    selected?: boolean;
  }

  /** @deprecated */
  interface FormEntryDependency {
    name: string;
    parent: string;
    transform?: Function;
  }

  /** @deprecated */
  interface MapTileLayerData {
    attribution: string;
    maxZoom?: number;
    token?: string;
    url: string;
  }

  /** @deprecated */
  type CarrierDeliveryDependencies = Record<MyParcel.CarrierName, DeliveryDependencies>;

  /** @deprecated */
  interface DeliveryDependencies {
    // ISO date string
    deliveryDate: Record<
      string,
      {
        deliveryMoment: DeliveryDependencyMoments;
      }
    >;
  }

  /** @deprecated */
  type DeliveryDependencyMoments = Record<DeliveryType, DeliveryDependencyMoment>;

  /** @deprecated */
  interface DeliveryDependencyMoment {
    moments: {
      start: string;
      end: string;
    };
    shipmentOptions: Record<MyParcel.ShipmentOptionName, EnumSchema<boolean>>;
  }

  /** @deprecated */
  interface DeliveryOptionsRequestParameters {
    carrier: MyParcel.CarrierIdentifier;
    cc: string;
    cutoff_time?: string;
    deliverydays_window?: number;
    dropoff_days?: string;
    dropoff_delay?: number;
    include?: 'shipment_options';
    monday_delivery?: boolean;
    number: number;
    package_type?: MyParcel.PackageType;
    platform: MyParcel.Platform;
    postal_code: string;
    saturday_delivery?: boolean;
  }
}

/** @deprecated */
declare module 'MyParcel' {
  export = MyParcel;
}

/** @deprecated */
declare module 'MyParcelDeliveryOptions' {
  export = MyParcelDeliveryOptions;
}
