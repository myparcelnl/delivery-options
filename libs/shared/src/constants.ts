import {PlatformName} from '@myparcel/constants';

export enum SubscriptionType {
  Required = 1,
  Never = 0,
  Optional = -1,
}

export const DEFAULT_PLATFORM = PlatformName.MyParcel;

export const FORM_NAME_DELIVERY_OPTIONS = 'deliveryOptions';

export enum CarrierSetting {
  AllowDeliveryOptions = 'allowDeliveryOptions',
  AllowEveningDelivery = 'allowEveningDelivery',
  AllowMondayDelivery = 'allowMondayDelivery',
  AllowMorningDelivery = 'allowMorningDelivery',
  AllowOnlyRecipient = 'allowOnlyRecipient',
  AllowPackageTypeDigitalStamp = 'allowPackageTypeDigitalStamp',
  AllowPackageTypeMailbox = 'allowPackageTypeMailbox',
  AllowPickupLocations = 'allowPickupLocations',
  AllowSameDayDelivery = 'allowSameDayDelivery',
  AllowSaturdayDelivery = 'allowSaturdayDelivery',
  /** @deprecated use ShowDeliveryDate instead */
  AllowShowDeliveryDate = 'allowShowDeliveryDate',
  AllowSignature = 'allowSignature',
  CutoffTime = 'cutoffTime',
  CutoffTimeSameDay = 'cutoffTimeSameDay',
  DeliveryDaysWindow = 'deliveryDaysWindow',
  DropOffDays = 'dropOffDays',
  DropOffDelay = 'dropOffDelay',
  FridayCutoffTime = 'fridayCutoffTime',
  PackageType = 'packageType',
  PriceEveningDelivery = 'priceEveningDelivery',
  PriceMondayDelivery = 'priceMondayDelivery',
  PriceMorningDelivery = 'priceMorningDelivery',
  PriceOnlyRecipient = 'priceOnlyRecipient',
  PricePackageTypeDigitalStamp = 'pricePackageTypeDigitalStamp',
  PricePackageTypeMailbox = 'pricePackageTypeMailbox',
  PricePickup = 'pricePickup',
  PriceSameDayDelivery = 'priceSameDayDelivery',
  PriceSaturdayDelivery = 'priceSaturdayDelivery',
  PriceSignature = 'priceSignature',
  PriceStandardDelivery = 'priceStandardDelivery',
  SaturdayCutoffTime = 'saturdayCutoffTime',
  ShowDeliveryDate = 'showDeliveryDate',
}

export enum ConfigSetting {
  ApiBaseUrl = 'apiBaseUrl',
  CarrierSettings = 'carrierSettings',
  Currency = 'currency',
  Locale = 'locale',
  PickupLocationsDefaultView = 'pickupLocationsDefaultView',
  PickupLocationsMapTileLayerData = 'pickupLocationsMapTileLayerData',
  PickupShowDistance = 'pickupShowDistance',
  Platform = 'platform',
  ShowDeliveryDate = 'showDeliveryDate',
  ShowPriceSurcharge = 'showPriceSurcharge',
  ShowPrices = 'showPrices',
}

export enum PickupLocationsView {
  Map = 'map',
  List = 'list',
}

export enum ComponentName {
  Button = 'Button',
  CheckboxGroup = 'CheckboxGroup',
  Checkbox = 'Checkbox',
  Radio = 'Radio',
  RadioGroup = 'RadioGroup',
  Select = 'Select',
  Text = 'Text',
}

export enum AddressField {
  Cc = 'cc',
  City = 'city',
  Number = 'number',
  PostalCode = 'postalCode',
  Street = 'street',
}

export enum OptionGroup {
  PackageType = 'packageType',
  Delivery = 'delivery',
  ShipmentOption = 'shipmentOption',
  Feature = 'feature',
  DropOff = 'dropOff',
}

export enum OptionType {
  String = 'string',
  Boolean = 'boolean',
  Number = 'number',
  Currency = 'currency',
  Date = 'date',
  Time = 'time',
  Select = 'select',
}

export const QUERY_KEY_CARRIERS = 'carriers';

export const QUERY_KEY_DELIVERY_OPTIONS = 'deliveryOptions';

export const QUERY_KEY_PICKUP_LOCATIONS = 'pickupLocations';
