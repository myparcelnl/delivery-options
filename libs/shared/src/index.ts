export * from './components';
export * from './composables';
export * from './constants';
export * from './data';
export * from './defineConfig';
export * from './legacy';
export * from './types';
export * from './validator';
