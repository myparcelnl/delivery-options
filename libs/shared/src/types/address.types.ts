import {type AddressField} from '../enums';

interface BaseAddress {
  [AddressField.Cc]: string;
  [AddressField.City]: string;
  [AddressField.PostalCode]: string;
}

export interface DeliveryOptionsOldAddress extends BaseAddress {
  [AddressField.Number]: string | number;
}

export interface DeliveryOptionsAddress extends BaseAddress {
  [AddressField.Street]: string;
}

export interface ResolvedDeliveryOptionsAddress extends BaseAddress {
  [AddressField.Street]: string;
}
