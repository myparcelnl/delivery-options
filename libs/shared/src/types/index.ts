export * from './address.types';
export * from './carriers.types';
export * from './common.types';
export * from './config.types';
export * from './element.types';
export * from './form.types';
export * from './input.types';
export * from './options.types';
export * from './platform.types';
export * from './query.types';
export * from './validator.types';
