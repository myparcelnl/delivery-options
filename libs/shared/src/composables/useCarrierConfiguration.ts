import {useMemoize} from '@vueuse/core';
import {type DeliveryTypeName, type PackageTypeName, type ShipmentOptionName} from '@myparcel/constants';
import {getPlatformConfig, resolveCarrierName} from '../utils';
import {type CarrierIdentifier, type SupportedPlatformName} from '../types';
import {useCarrier} from '../sdk';

const getCarrierConfiguration = useMemoize((carrierIdentifier: CarrierIdentifier, platform?: SupportedPlatformName) => {
  const platformConfig = getPlatformConfig(platform);
  const carrierName = resolveCarrierName(carrierIdentifier);

  const foundCarrier = platformConfig?.carriers.find((carrier) => carrier.name === carrierName);

  if (!foundCarrier) {
    throw new Error(`No configuration found for carrier ${carrierIdentifier}`);
  }

  return foundCarrier;
});

export interface UseCarrierConfiguration {
  carrier: ReturnType<typeof useCarrier>;
  config: ReturnType<typeof getCarrierConfiguration>;

  hasDeliveryType(deliveryType: DeliveryTypeName): boolean;

  hasFeature(feature: string): boolean;

  hasPackageType(packageType: PackageTypeName): boolean;

  hasShipmentOption(shipmentOption: ShipmentOptionName): boolean;
}

export const useCarrierConfiguration = (
  carrierIdentifier: CarrierIdentifier,
  platform?: SupportedPlatformName,
): UseCarrierConfiguration => {
  const config = getCarrierConfiguration(carrierIdentifier, platform);
  const carrier = useCarrier(carrierIdentifier);

  return {
    carrier,
    config,

    hasPackageType(packageType: PackageTypeName) {
      return config.packageTypes?.includes(packageType) ?? false;
    },

    hasDeliveryType(deliveryType: DeliveryTypeName) {
      return config.deliveryTypes?.includes(deliveryType) ?? false;
    },

    hasShipmentOption(shipmentOption: ShipmentOptionName) {
      return config.shipmentOptions?.includes(shipmentOption) ?? false;
    },

    hasFeature(feature: string) {
      return config.features?.includes(feature) ?? false;
    },
  };
};
