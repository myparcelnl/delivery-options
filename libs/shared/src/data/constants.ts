import {DeliveryTypeName, PackageTypeName, PlatformName, ShipmentOptionName} from '@myparcel/constants';
import {type DropOffEntryObject} from '../types';
import {CarrierSetting} from './enums';

export const SUPPORTED_PLATFORMS = [PlatformName.MyParcel, PlatformName.SendMyParcel] as const;

export const SUPPORTED_DELIVERY_TYPES = [
  DeliveryTypeName.Standard,
  DeliveryTypeName.Evening,
  DeliveryTypeName.Morning,
  DeliveryTypeName.Pickup,
] as const;

export const SUPPORTED_SHIPMENT_OPTIONS = [ShipmentOptionName.OnlyRecipient, ShipmentOptionName.Signature] as const;

export const SUPPORTED_PACKAGE_TYPES = [
  PackageTypeName.Package,
  PackageTypeName.Mailbox,
  PackageTypeName.DigitalStamp,
  PackageTypeName.PackageSmall,
] as const;

export const DEFAULT_PLATFORM = PlatformName.MyParcel;

export const FORM_NAME_DELIVERY_OPTIONS = 'deliveryOptions';

export const REQUEST_KEY_CARRIERS = 'carriers';

export const REQUEST_KEY_DELIVERY_OPTIONS = 'deliveryOptions';

export const REQUEST_KEY_PICKUP_LOCATIONS = 'pickupLocations';

export const DROP_OFF_WEEKDAY = 'weekday' satisfies keyof DropOffEntryObject;

export const DROP_OFF_CUTOFF_TIME = CarrierSetting.CutoffTime satisfies keyof DropOffEntryObject;

export const DROP_OFF_SAME_DAY_CUTOFF_TIME = CarrierSetting.CutoffTimeSameDay satisfies keyof DropOffEntryObject;

export const DAY_MONDAY = 1;

export const DAY_TUESDAY = 2;

export const DAY_WEDNESDAY = 3;

export const DAY_THURSDAY = 4;

export const DAY_FRIDAY = 5;

export const DAY_SATURDAY = 6;

export const DAY_SUNDAY = 0;

export const DAYS_IN_WEEK = 7;

export const NBSP = '\u00A0';

export const CARRIER_IDENTIFIER_SEPARATOR = ':';

export const API_DATE_FORMAT = 'y-MM-dd kk:mm:ss';

export const DEFAULT_TIMEZONE_TYPE = 3;

export const DEFAULT_TIMEZONE = 'Europe/Amsterdam';

export const DELIVERY_TIMEFRAME_TYPE_START = 'start';

export const DELIVERY_TIMEFRAME_TYPE_END = 'end';
