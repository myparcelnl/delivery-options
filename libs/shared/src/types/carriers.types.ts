import {type Carrier} from '@myparcel/sdk';
import {type CarrierName} from '@myparcel/constants';
import {
  type SupportedDeliveryTypeName,
  type SupportedPackageTypeName,
  type SupportedShipmentOptionName,
} from './platform.types';
import {type CarrierIdentifier, type ConfigKey} from './config.types';

export interface CarrierObject {
  identifier: CarrierIdentifier;
  name: CarrierName;
}

export interface CarrierWithIdentifier extends Carrier {
  identifier: CarrierIdentifier;
}

export type FullCarrier = CarrierWithIdentifier & {
  hasDeliveryInCountry(countryCode: string): boolean;
  hasDeliveryType(deliveryType: SupportedDeliveryTypeName): boolean;
  hasFeature(feature: ConfigKey): boolean;
  hasPackageType(packageType: SupportedPackageTypeName): boolean;
  hasPickupInCountry(countryCode: string): boolean;
  hasShipmentOption(shipmentOption: SupportedShipmentOptionName): boolean;
};
