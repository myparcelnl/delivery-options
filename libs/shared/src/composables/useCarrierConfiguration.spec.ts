import {describe, expect, it} from 'vitest';
import {get} from '@vueuse/core';
import {useCarrier} from '@myparcel-do/shared';
import {CarrierName, PlatformName} from '@myparcel/constants';
import {type CarrierIdentifier, type SupportedPlatformName} from '../types';
import {type UseCarrierConfiguration, useCarrierConfiguration} from './useCarrierConfiguration';

const allPlatforms = [PlatformName.MyParcel, PlatformName.SendMyParcel] satisfies SupportedPlatformName[];
const allCarriers = Object.values(CarrierName);

const matrix = allPlatforms.flatMap((platform) => {
  return allCarriers.map((carrier) => [carrier, platform]);
}) as [CarrierName, SupportedPlatformName][];

describe('useCarrierConfiguration', () => {
  it.each(matrix)('returns the carrier configuration for "%s" on "%s"', (carrierName, platformName) => {
    let result: string | UseCarrierConfiguration['config'];

    try {
      const configuration = useCarrierConfiguration(carrierName, platformName);

      result = configuration.config;
    } catch (error) {
      // Not available on this platform
      result = '❌';
    }

    expect(result).toMatchSnapshot();
  });

  it('throws an error when a carrier is not found', () => {
    expect(() => useCarrierConfiguration('foo' as CarrierIdentifier)).toThrow('No configuration found for carrier foo');
  });

  it('resolves carrier by identifier', async () => {
    const configuration = useCarrierConfiguration(`${CarrierName.PostNl}:12345`);

    // Wait for the carrier to load its data
    await useCarrier(CarrierName.PostNl).load();

    expect(get(configuration.carrier.data)?.name).toBe(CarrierName.PostNl);
  });
});
