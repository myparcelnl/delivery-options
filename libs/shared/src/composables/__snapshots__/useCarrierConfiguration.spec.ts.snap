// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useCarrierConfiguration > returns the carrier configuration for "bol.com" on "belgie" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "bol.com" on "belgie" 2`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "bol.com" on "myparcel" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "bol.com" on "myparcel" 2`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "bpost" on "belgie" 1`] = `
{
  "addressFields": [
    "postalCode",
    "street",
    "city",
  ],
  "deliveryCountries": [
    "BE",
    "NL",
  ],
  "deliveryTypes": [
    "standard",
  ],
  "features": [
    [
      "showDeliveryDate",
    ],
    [
      "deliveryDaysWindow",
    ],
    [
      "dropOffDays",
    ],
    [
      "dropOffDelay",
    ],
    [
      "cutoffTime",
    ],
  ],
  "name": "bpost",
  "packageTypes": [
    "package",
  ],
  "pickupCountries": [
    "BE",
    "NL",
  ],
  "shipmentOptions": [
    "only_recipient",
    "signature",
  ],
  "subscription": 0,
}
`;

exports[`useCarrierConfiguration > returns the carrier configuration for "bpost" on "myparcel" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "cheapcargo" on "belgie" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "cheapcargo" on "myparcel" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "dhl" on "belgie" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "dhl" on "myparcel" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "dhleuroplus" on "belgie" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "dhleuroplus" on "myparcel" 1`] = `
{
  "deliveryCountries": [
    "BE",
    "BG",
    "DK",
    "DE",
    "EE",
    "FI",
    "FR",
    "GR",
    "HU",
    "IE",
    "IT",
    "HR",
    "LV",
    "LT",
    "LU",
    "NL",
    "AT",
    "PL",
    "PT",
    "RO",
    "SI",
    "SK",
    "ES",
    "CZ",
    "SE",
    "GB",
  ],
  "deliveryTypes": [
    "standard",
  ],
  "features": [],
  "name": "dhleuroplus",
  "packageTypes": [
    "package",
  ],
  "subscription": -1,
}
`;

exports[`useCarrierConfiguration > returns the carrier configuration for "dhlforyou" on "belgie" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "dhlforyou" on "myparcel" 1`] = `
{
  "addressFields": [
    "city",
    "postalCode",
  ],
  "deliveryCountries": [
    "NL",
    "BE",
  ],
  "deliveryTypes": [
    "standard",
  ],
  "features": [
    [
      "cutoffTime",
    ],
    [
      "deliveryDaysWindow",
    ],
    [
      "dropOffDays",
    ],
    [
      "dropOffDelay",
    ],
  ],
  "name": "dhlforyou",
  "packageTypes": [
    "package",
  ],
  "pickupCountries": [
    "NL",
  ],
  "shipmentOptions": [
    "only_recipient",
    "same_day_delivery",
    "signature",
  ],
  "subscription": -1,
}
`;

exports[`useCarrierConfiguration > returns the carrier configuration for "dhlparcelconnect" on "belgie" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "dhlparcelconnect" on "myparcel" 1`] = `
{
  "deliveryCountries": [
    "BG",
    "DE",
    "EE",
    "FI",
    "GR",
    "HU",
    "IE",
    "IT",
    "HR",
    "LV",
    "LT",
    "LU",
    "AT",
    "PL",
    "PT",
    "RO",
    "SI",
    "SK",
    "ES",
    "CZ",
  ],
  "deliveryTypes": [
    "standard",
    "pickup",
  ],
  "features": [],
  "name": "dhlparcelconnect",
  "packageTypes": [
    "package",
  ],
  "pickupCountries": [
    "BG",
    "DK",
    "DE",
    "EE",
    "FI",
    "FR",
    "GR",
    "HU",
    "IE",
    "IT",
    "HR",
    "LV",
    "LT",
    "LU",
    "AT",
    "PL",
    "PT",
    "RO",
    "SI",
    "SK",
    "ES",
    "CZ",
    "SE",
    "GB",
  ],
  "subscription": -1,
}
`;

exports[`useCarrierConfiguration > returns the carrier configuration for "dpd" on "belgie" 1`] = `
{
  "addressFields": [
    "postalCode",
    "street",
    "city",
  ],
  "deliveryCountries": [
    "AT",
    "BE",
    "BG",
    "CZ",
    "DK",
    "EE",
    "FI",
    "FR",
    "DE",
    "GR",
    "HU",
    "IE",
    "IT",
    "LV",
    "LI",
    "LT",
    "LU",
    "NL",
    "PL",
    "PT",
    "RO",
    "SK",
    "SI",
    "ES",
    "SE",
  ],
  "deliveryTypes": [
    "standard",
    "pickup",
  ],
  "features": [
    [
      "dropOffDays",
    ],
    [
      "dropOffDelay",
    ],
    [
      "cutoffTime",
    ],
  ],
  "name": "dpd",
  "packageTypes": [
    "package",
  ],
  "pickupCountries": [
    "AT",
    "BE",
    "CZ",
    "DK",
    "EE",
    "FI",
    "FR",
    "DE",
    "HU",
    "LV",
    "LT",
    "LU",
    "NL",
    "PL",
    "PT",
    "SK",
    "SI",
    "ES",
    "GB",
  ],
  "subscription": 0,
}
`;

exports[`useCarrierConfiguration > returns the carrier configuration for "dpd" on "myparcel" 1`] = `
{
  "addressFields": [
    "postalCode",
    "street",
    "city",
  ],
  "deliveryCountries": [
    "AT",
    "BE",
    "BG",
    "CZ",
    "DK",
    "EE",
    "FI",
    "FR",
    "DE",
    "GR",
    "HU",
    "IE",
    "IT",
    "LV",
    "LI",
    "LT",
    "LU",
    "NL",
    "PL",
    "PT",
    "RO",
    "SK",
    "SI",
    "ES",
    "SE",
  ],
  "deliveryTypes": [
    "standard",
    "pickup",
  ],
  "features": [
    [
      "dropOffDays",
    ],
    [
      "dropOffDelay",
    ],
    [
      "cutoffTime",
    ],
  ],
  "name": "dpd",
  "packageTypes": [
    "package",
  ],
  "pickupCountries": [
    "AT",
    "BE",
    "CZ",
    "DK",
    "EE",
    "FI",
    "FR",
    "DE",
    "HU",
    "LV",
    "LT",
    "LU",
    "NL",
    "PL",
    "PT",
    "SK",
    "SI",
    "ES",
    "GB",
  ],
  "subscription": 1,
}
`;

exports[`useCarrierConfiguration > returns the carrier configuration for "instabox" on "belgie" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "instabox" on "myparcel" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "postnl" on "belgie" 1`] = `
{
  "deliveryCountries": [
    "BE",
    "NL",
  ],
  "deliveryTypes": [
    "standard",
  ],
  "features": [
    [
      "showDeliveryDate",
    ],
    [
      "deliveryDaysWindow",
    ],
    [
      "dropOffDays",
    ],
    [
      "dropOffDelay",
    ],
    [
      "cutoffTime",
    ],
  ],
  "name": "postnl",
  "packageTypes": [
    "package",
  ],
  "pickupCountries": [
    "BE",
    "NL",
  ],
  "shipmentOptions": [
    "signature",
  ],
  "subscription": 0,
}
`;

exports[`useCarrierConfiguration > returns the carrier configuration for "postnl" on "myparcel" 1`] = `
{
  "addressFields": [
    "postalCode",
    "street",
    "city",
  ],
  "deliveryCountries": [
    "NL",
    "BE",
  ],
  "deliveryTypes": [
    "standard",
    "morning",
    "evening",
    "pickup",
  ],
  "features": [
    [
      "showDeliveryDate",
    ],
    [
      "deliveryDaysWindow",
    ],
    [
      "dropOffDays",
    ],
    [
      "dropOffDelay",
    ],
    [
      "cutoffTime",
    ],
  ],
  "name": "postnl",
  "packageTypes": [
    "package",
    "mailbox",
    "digital_stamp",
    "letter",
  ],
  "pickupCountries": [
    "NL",
    "BE",
  ],
  "shipmentOptions": [
    "only_recipient",
    "signature",
  ],
  "subscription": -1,
}
`;

exports[`useCarrierConfiguration > returns the carrier configuration for "ups" on "belgie" 1`] = `"❌"`;

exports[`useCarrierConfiguration > returns the carrier configuration for "ups" on "myparcel" 1`] = `
{
  "addressFields": [
    "postalCode",
    "street",
    "city",
  ],
  "deliveryCountries": [
    "BG",
    "DE",
    "EE",
    "FI",
    "GR",
    "HU",
    "IE",
    "IT",
    "HR",
    "LV",
    "LT",
    "LU",
    "AT",
    "PL",
    "PT",
    "RO",
    "SI",
    "SK",
    "ES",
    "CZ",
  ],
  "deliveryTypes": [
    "standard",
  ],
  "fakeDelivery": true,
  "name": "ups",
  "packageTypes": [
    "package",
  ],
  "pickupCountries": [
    "DE",
  ],
  "subscription": 1,
}
`;
